types/es-abstract/es2018.d.ts

```diff
@@ -0,0 +1,47 @@
+import ES2017 = require('./es2017');
+
+type PromiseConstructorLikeReturnType<C extends PromiseConstructorLike, T> = C extends new (
+	executor: (resolve: (value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void,
+) => infer R
+	? R
+	: PromiseLike<T>;
+
+interface ES2018 extends Omit<ES2017, 'EnumerableOwnProperties' | 'GetSubstitution' | 'IsPropertyDescriptor'> {
+	readonly EnumerableOwnPropertyNames: typeof ES2017.EnumerableOwnProperties;
+
+	// tslint:disable-next-line: ban-types
+	thisSymbolValue(value: symbol | Symbol): symbol;
+
+	IsStringPrefix(p: string, q: string): boolean;
+	NumberToString(m: number): string;
+	CopyDataProperties<T, S, E extends ES2017.PropertyKey>(target: T, source: S, excludedItems: E[]): T & Omit<S, E>;
+	PromiseResolve<T>(C: PromiseConstructor, x: T): Promise<T>;
```

Comment:
```suggestion
	PromiseResolve: typeof Promise.resolve
```

Reply:
No, because `Promise.resolve` takes one parameter, whereas this takes 2 as it’s essentially:
```js
function PromiseResolve(C, x) {
	return Promise.resolve.call(C, x);
}
```

Reply:
ahhh fair point. in that case, what about `typeof Promise.resolve.call`?

Reply:
That won’t work, as `typeof Promise.resolve.call` is:
```ts
(this: Function, thisArg: any, ...argArray: any[]) => any
```

Reply:
hmm, i thought the .call overloads were supposed to work here. `Promise.resolve.call` should  have a typeof `(thisArg, value)`.

Reply:
That’s only when you use `strictBindCallApply`, and only for code that actually uses it like so:
```ts
function NewCustomPromise<T>(someValue: T): CustomPromise<T> {
	const p = Promise.resolve.call(CustomPromise, someValue);
	// Do stuff with `p`
	return p;
}
```